server/controllers/admin/requestStatus.ts

```diff
@@ -0,0 +1,23 @@
+import { Request, Response, NextFunction } from 'express';
+
+import requestStatusQuery from '../../queries';
+import { CustomError } from '../../utils';
+import requestStatusSchema from '../../validation/requestStatusSchema';
+
+const requestStatus = async (req : Request, res : Response, next : NextFunction) => {
+  try {
+    const { status } = req.body;
+    const { requestId } = req.params;
+    const isValid = await requestStatusSchema.validateAsync({ status, requestId });
+    if (isValid) {
+      await requestStatusQuery(status, +requestId);
+      res.json({ msg: 'successful' });
+    } else {
+      throw new CustomError(400, 'invalid input');
+    }
```

Comment:
If the data is not validated, then the validation function will throw the error and your catch will catch it. 
so if you are still in line 12 then your data is validated, which means no need to isValid and it doesn't return a boolean.